### ISSUE TITLE
Refactor code using Biryani

### ISSUE THREAD

**Issue by maukoquiroga** [openfisca/openfisca-core#691]
As it is no longer maintained, we should either :

- replace it for other specific libraries doing currying, type checking and conversions separately
- refactor code to do python-style composition
- avoid composition

Note that most code using it is untested.

Connected to openfisca/openfisca-core#716
Connected to #540

**Comment by fpagnoux** [comment]
FI, we tried to remove dependencies to Biryani each time we worked on a module since late 2016. 

While the Byriani is intellectually seducing, practically it was really hard to debug and find what's going on. We replaced it with basic python code, which can be a little tedious, but allows us to finely set our error message.

I think the last Byriani stronghold is the scenarios, which are used to parse tests.
